code/datums/status_effects/debuffs/debuffs.dm

```diff
@@ -958,5 +958,74 @@
 /datum/movespeed_modifier/careful_driving
 	multiplicative_slowdown = 3
 
+/datum/status_effect/midas_blight
+	id = "midas_blight"
+	alert_type = /atom/movable/screen/alert/status_effect/midas_blight
+	status_type = STATUS_EFFECT_REPLACE
+	remove_on_fullheal = TRUE
+	processing_speed = STATUS_EFFECT_FAST_PROCESS
+	/// The visual overlay, helps tell both you and enemies how much gold is in your system
+	var/mutable_appearance/midas_overlay
+
+/datum/status_effect/midas_blight/on_creation(mob/living/new_owner, duration = 1)
+	src.duration = duration
+	new_owner.visible_message("[src.duration]")
+	return ..()
+
+/atom/movable/screen/alert/status_effect/midas_blight
+	name = "Midas Blight"
+	desc = "Your blood is being turned to gold, slowing your movements!"
+	icon_state = "midas_blight"
+
+/datum/status_effect/midas_blight/on_apply()
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/midas_blight/soft, update = TRUE)
+	midas_overlay = mutable_appearance('icons/mob/effects/debuff_overlays.dmi', "midas_1")
+	midas_overlay.blend_mode = BLEND_MULTIPLY
+	owner.add_overlay(midas_overlay)
```

Comment:
Instead of adding overlay directly, you might want to hook onto owner `update_overlay` signal and add it there.

Reply:
I'm not exactly sure how I'd do that. If I hooked it to the `update_overlay` signal then wouldn't it repeatedly place the overlay onto owner every time `update_overlay` is called, making it stack?